contrib/opencensus-ext-azure/opencensus/ext/azure/metrics_exporter/__init__.py

```diff
@@ -12,31 +12,89 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-import threading
+import time
 
 from opencensus.ext.azure.common import Options
+from opencensus.metrics.export import value
+from opencensus.metrics.export.metric_descriptor import MetricDescriptorType
 from opencensus.metrics import transport
 from opencensus.stats import stats
+from opencensus.ext.azure.common import utils
+from opencensus.ext.azure.common.protocol import Data
+from opencensus.ext.azure.common.protocol import DataPoint
+from opencensus.ext.azure.common.protocol import Envelope
+from opencensus.ext.azure.common.protocol import MetricData
+from opencensus.ext.azure.common.transport import TransportMixin
 
 __all__ = ['MetricsExporter', 'new_metrics_exporter']
 
 
-class MetricsExporter(object):
+class MetricsExporter(TransportMixin):
     """Metrics exporter for Microsoft Azure Monitor."""
 
-    def __init__(self, options):
+    def __init__(self, options=None):
```

Comment:
Is there a reason to prefer `options` other than `**options`?

Reply:
The users should create metric exporter instances via new_metrics_exporter(**options) instead of calling the initializer directly. This is a design choice, as I want the starting of the exporter thread to be done outside.

Reply:
Got it.